src/k8s-engine/client.ts

```diff
@@ -518,11 +530,19 @@ export class KubeClient extends events.EventEmitter {
    * @param port The port to forward to on the endpoint.
    */
   async cancelForwardPort(namespace: string, endpoint: string, port: number | string) {
+    const targetName = this.targetName(namespace, endpoint, port);
     const server = this.servers.get(namespace, endpoint, port);
 
     this.servers.delete(namespace, endpoint, port);
     if (server) {
-      await new Promise(resolve => server.close(resolve));
+      await new Promise((resolve) => {
+        server.close(resolve);
```

Comment:
Shouldn't this be at the end of the callback (so that we don't call `resolve` too early)?

Reply:
`server.close()` won't invoke the callback until all connections are closed, so we won't resolve early here. 

It's a common pattern pattern first call `server.close()` because this will stop accepting new connections before cleaning up sockets. In other words, there's the possibility that we would receive new connections after destroying sockets if we swapped the order.